drivers/dac/ltc2672/ltc2672.h

```diff
@@ -0,0 +1,257 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#ifndef __MAX31855_H__
+#define __MAX31855_H__
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdint.h>
+#include <stdbool.h>
+#include "no_os_spi.h"
+#include "no_os_error.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+/* LTC2672 Masks */
+#define LTC2672_16_DONT_CARE 	0xFFF0
+#define LTC2672_MUX_DONT_CARE 	0xFFFE0
+#define LTC2672_DUMMY		 	0xFFFF
+
+/* LTC2672 Constants */
+#define LTC2672_BASE_CURRENT 3.125 // base current in mA
+#define LTC2672_16BIT_RESO   65535
+#define LTC2672_12BIT_RESO   4095
+
+enum ltc2672_commands {
+	LTC2672_CODE_TO_CHANNEL_X,
+	LTC2672_PWRUP_UPD_CHANNEL_X,
+	LTC2672_CODE_TO_CHANNEL_X_PWRUP_UPD_CHANNEL_ALL,
+	LTC2672_CODE_PWRUP_UPD_CHANNEL_X,
+	LTC2672_PWRDWN_CHANNEL_X,
+	LTC2672_PWRDWN_DEV,
+	LTC2672_SPAN_TO_CHANNEL_X,
+	LTC2672_CNFG_CMD,
+	LTC2672_CODE_TO_CHANNEL_ALL,
+	LTC2672_PWRUP_UPD_CHANNEL_ALL,
+	LTC2672_CODE_PWRUP_UPD_CHANNEL_ALL,
+	LTC2672_MON_MUX,
+	LTC2672_TOGGLE_SEL,
+	LTC2672_TOGGLE_GLBL,
+	LTC2672_SPAN_TO_CHANNEL_ALL,
+	LTC2672_NO_OP
+};
+
+/* LTC2672 Command Generation */
+#define LTC2672_COMMAND32_GENERATE(comm, add, dat) \
+		(0xFF << 24) | (comm << 20) | (add << 16) | (dat)
+
+#define LTC2672_COMMAND24_GENERATE(comm, add, dat) \
+		(comm << 20) | (add << 16) | (dat)
+
+#define LTC2672_SPAN_SET(span_code) 	 LTC2672_16_DONT_CARE | span_code
+
+#define LTC2672_MUX_SET(mux_code) 	 	 LTC2672_MUX_DONT_CARE | mux_code
+
+#define LTC2672_MUX32_GENERATE(comm, dat) \
+		(0xFF << 24) | (comm << 20) | (dat)
+
+#define LTC2672_MUX24_GENERATE(comm, dat) \
+		(comm << 20) | (dat)
+
+/* Device Family */
+enum ltc2672_device_id {
+	LTC2672_12 = 0,
+	LTC2672_16 = 1
+};
+
+/* DAC Channels */
+enum ltc2672_dac_ch {
+	LTC2672_DAC0,
+	LTC2672_DAC1,
+	LTC2672_DAC2,
+	LTC2672_DAC3,
+	LTC2672_DAC4
+};
+
+/* Output Range */
+enum ltc2672_out_range {
+	LTC2672_OFF, // Off mode
+	LTC2672_50VREF, // 3.125mA
+	LTC2672_100VREF, // 6.25mA
+	LTC2672_200VREF, //12.5mA
+	LTC2672_400VREF, // 25mA
+	LTC2672_800VREF, // 50mA
+	LTC2672_1600VREF, // 100mA
+	LTC2672_3200VREF,  // 200mA
+	LTC2672_VMINUS_VREF,
+	LTC2672_4800VREF = 0XF // 300mA
+};
+
+/* Multiplexer Command Codes */
+enum ltc2672_mux_commands {
+	LTC2672_MUX_DISABLED,
+	LTC2672_MUX_IOUT0,
+	LTC2672_MUX_IOUT1,
+	LTC2672_MUX_IOUT2,
+	LTC2672_MUX_IOUT3,
+	LTC2672_MUX_IOUT4,
+	LTC2672_MUC_VCC,
+	LTC2672_MUX_VREF = 0x08,
```

Comment:
in this case you either initialize all members of the enum, or none of them.

Reply:
It's weird to specify a value for all of them. Just specify the value for those that create a gap like LTC2672_MUX_VREF which is +2 compared to  previous one LTC2672_MUC_VCC and leave the others without explicit values.

Reply:
So I think @jsanbuen was right.